Sources/TSCclibc/process.c

```diff
@@ -32,4 +36,61 @@ bool SPM_posix_spawn_file_actions_addchdir_np_supported() {
 #endif
 }
 
+int SPM_fork_exec_chdir(pid_t *pid, const char *cwd,
+                        const char *cmd, char *const argv[], char *const envp[],
+                        int in_pipe[], int out_pipe[], int err_pipe[], bool redirect_out, bool redirect_err) {
+    *pid = fork();
+
+    if (*pid < 0) {
+        perror("fork() failed");
+        _exit(EXIT_FAILURE);
+    } else if (*pid > 0) { // Parent process
+        // Wait for child process to finish
+        int status;
+        waitpid(*pid, &status, 0);
+        return status;
+    } else { // Child process
+        // Change working directory then execute cmd
+        if (chdir(cwd)) {
+            _exit(EXIT_FAILURE);
+        }
+        
+        // Replicate pipe logic in TSCBasic.Process.launch()
+        
+        // Dupe the read portion of the remote to 0.
+        dup2(in_pipe[0], 0);
+        // Close the other side's pipe since it was duped to 0.
+        close(in_pipe[0]);
+        close(in_pipe[1]);
+        
+        if (redirect_out) {
+            // Open the write end of the pipe.
+            dup2(out_pipe[1], 1);
```

Comment:
who's reading that pipe? It appears like the parent waits for the child to complete and _then_ reads those pipes.

That's a deadlock.

The child will fill up the pipe until it's full (assuming the child writes enough). And then the child will block until something reads from that pipe. So the child won't exit but the parent will wait for it to exit:

child waits for parent to read <-> parent waits for child to exit